portal-api/src/v1/users/routes.js

```diff
@@ -154,32 +155,44 @@ module.exports.findById = (req, res, next) => {
 };
 
 module.exports.updateById = (req, res, next) => {
-  findOne(req.params._id, (error, user) => {
-    if (error) {
-      next(error);
-    } else if (user) {
+  try {
+    const userIsAdmin = req.user?.roles?.includes(ADMIN);
+    const userIsChangingTheirOwnPassword = req.user?._id?.toString() === req.params._id
+    && !Object.keys(req.body).some(
+      (property) => !['password', 'passwordConfirm', 'currentPassword'].includes(property)
+    );
+    if (!userIsAdmin && !userIsChangingTheirOwnPassword) {
+      return res.status(403).send();
+    }
+    return findOne(req.params._id, (error, user) => {
+      if (error) {
+        return next(error);
+      }
+      if (!user) {
+        return res.status(404).send();
+      }
       const errors = applyUpdateRules(user, req.body);
       if (errors.length) {
-        res.status(400).json({
+        return res.status(400).json({
           success: false,
           errors: {
             count: errors.length,
             errorList: combineErrors(errors),
           },
         });
-      } else {
-        update(req.params._id, req.body, (updateErr, updatedUser) => {
-          if (updateErr) {
-            next(updateErr);
-          } else {
-            res.status(200).json(sanitizeUser(updatedUser));
-          }
-        });
       }
-    } else {
-      res.status(200).json({});
-    }
-  });
+
+      return update(req.params._id, req.body, (updateErr, updatedUser) => {
+        if (updateErr) {
+          return next(updateErr);
+        }
+        return res.status(200).json(sanitizeUser(updatedUser));
+      });
+    });
+  } catch (e) {
+    console.error(e);
```

Comment:
Can we do console.error('Error updating user', e) please? makes it easier to read in logs